src/quick-lint-js/parse.h

```diff
@@ -2495,6 +2512,45 @@ class parser {
     }
   }
 
+  std::optional<function_attributes> try_parse_function_with_leading_star() {
```

Comment:
The semantics of this function are a bit weird, so I suggest including a doc comment.

* If the function returns nullopt, no token is consumed
* If the function returns a `function_attributes`, tokens are consumed until the `kw_function` (i.e. the `*` and possibly a following `async` are skipped)

Also, including examples in the doc comment would be nice. (I notice that most functions don't have a doc comment with examples. Maybe we should change that...)

Reply:
> Maybe we should change that

Yes